server/src/main/java/io/druid/client/HttpServerInventoryView.java

```diff
@@ -328,8 +331,42 @@ public void run()
   //segmentViewInitialized on all registered segment callbacks.
   private void serverInventoryInitialized()
   {
+    long start = System.currentTimeMillis();
+    long serverSyncWaitTimeout = config.getServerTimeout() + 2 * ChangeRequestHttpSyncer.HTTP_TIMEOUT_EXTRA_MS;
+
+    List<DruidServerHolder> uninitializedServers = new ArrayList<>();
     for (DruidServerHolder server : servers.values()) {
-      server.awaitInitialization();
+      if (!server.isSyncedSuccessfullyAtleastOnce()) {
+        uninitializedServers.add(server);
+      }
+    }
+
+    while (!uninitializedServers.isEmpty() && ((System.currentTimeMillis() - start) < serverSyncWaitTimeout)) {
+      try {
+        Thread.sleep(5000);
```

Comment:
nit: use a constant for 5000?

Reply:
this place would be the only user of that variable. Given the context it is self explanatory so let us keep it hard coded.